### PR TITLE
Net various changes

### PR DESCRIPTION
**Pull request by tbursztyka**
These are non-critical changes (one optimization about icmpv6 though)

### CODE REVIEW

File: include/net/net_pkt.h [hunk 1]
```diff
@@ -419,12 +419,17 @@ static inline void net_pkt_set_priority(struct net_pkt *pkt,
...
+#else /* NET_TC_COUNT == 1 */
+#define net_pkt_set_priority(...)
+
 static inline u8_t net_pkt_priority(struct net_pkt *pkt)
 {
 	return 0;
 }
-#endif
+
+#define net_pkt_set_priority(...)
```
Comment (pfalcon) [inline include/net/net_pkt.h]: Why repeated twice? Why not use real params instead of "..."?

Comment (jukkar) [reply include/net/net_pkt.h]: Both ways can be used here. The benefit of ... is that if the number of parameters change in that func, we do not need to change the macro.

Comment (jukkar) [reply include/net/net_pkt.h]: > Why repeated twice?

That is actually correct, the net_pkt_set_priority() is set twice here.

Comment (pfalcon) [reply include/net/net_pkt.h]: > The benefit of ... 

Ok, if it's common pattern, sounds good, somehow I didn't pay attention to that.

Comment (pfalcon) [reply include/net/net_pkt.h]: > That is actually correct, the net_pkt_set_priority() is set twice here.

Why? Why have stuff like:

~~~
#define net_pkt_set_priority(...)
// foo
#define net_pkt_set_priority(...)
~~~

Comment (tbursztyka) [inline include/net/net_pkt.h]: my mistake, I had put the macro before the inline one, then realized the order is reversed (when there are 2 inilnes), so I did reverse the order... and forget to remove the first.

**Review by pfalcon — APPROVED** [review]

File: include/net/net_pkt.h (hunk 1 above)
Comment (jukkar) [reply include/net/net_pkt.h]: > That is actually correct, the net_pkt_set_priority() is set twice here.
> 
> Why? Why have stuff like:

I meant your comment is correct, not the actual code :)

**Review by jukkar — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**